sklearn/cluster/affinity_propagation_.py

```diff
@@ -288,17 +295,21 @@ def fit(self, X, y=None):
         if self.affinity == "precomputed":
             self.affinity_matrix_ = X
         elif self.affinity == "euclidean":
-            self.affinity_matrix_ = -euclidean_distances(X, squared=True)
+            self.affinity_matrix_ = -euclidean_distances(X, squared=False)
```

Comment:
I was referring to this change. If this makes the results better, we should change the behavior. But we can't  change is this abruptly.